scargo/commands/check.py

```diff
@@ -41,171 +44,179 @@ def scargo_check(
         with open(file_path, encoding="utf-8") as file:
             print(file_path)
             for line in file.readlines():
                 if "#pragma once" in line:
-                    found_pragma_once = True
-                    break
-
-        if not found_pragma_once:
-            error_counter += 1
-            logger.warning("Missing pragma in %s", file_path)
-
-            if fix_errors:
-                logger.info("Fixing...")
-
-                with open(file_path, encoding="utf-8") as file:
-                    old = file.read()
-
-                with open(file_path, "w", encoding="utf-8") as file:
-                    file.write("#pragma once\n")
-                    file.write("\n")
-                    file.write(old)
-
-    if fix_errors:
-        logger.info("Finished pragma check. Fixed problems in %s files.", error_counter)
-    else:
-        logger.info("Finished pragma check. Found problems in %s files.", error_counter)
-        if error_counter > 0:
-            logger.error("pragma check fail!")
-            sys.exit(1)
+                    return CheckResult(0)
+        logger.warning("Missing pragma in %s", file_path)
+        return CheckResult(1)
 
+    def fix_file(self, file_path: Path) -> None:
+        with open(file_path, encoding="utf-8") as file:
+            old = file.read()
 
-def check_copyright(config: Config, fix_errors: bool) -> None:
-    """
-    Private function used in commands `scargo check` and `scargo fix`.
+        with open(file_path, "w", encoding="utf-8") as file:
+            file.write("#pragma once\n")
+            file.write("\n")
+            file.write(old)
 
-    :param Config config: project configuration
-    :param bool fix_errors: if fix errors during code checking
-    :return:
-    """
-    logger = get_logger()
-    logger.info("Starting copyright check...")
 
-    error_counter = 0
+class CopyrightChecker(CheckerFixer):
+    check_name = "copyright"
+    can_fix = True
 
-    copyright_desc = config.check.copyright.description
-    if not copyright_desc:
-        logger.warning("No copyrights in defined in toml")
-        return
+    def check(self) -> None:
+        copyright_desc = self.get_check_config().description
+        if not copyright_desc:
+            logger.warning("No copyrights in defined in toml")
+            return
+        super().check()
 
-    for file_path in find_files(config.check.copyright.exclude, config):
-        found_copyright = False
-        any_copyright = False
+    def check_file(self, file_path: Path) -> CheckResult:
+        copyright_desc = self.get_check_config().description
+        assert copyright_desc
 
         with open(file_path, encoding="utf-8") as file:
             for line in file.readlines():
                 if copyright_desc in line:
-                    found_copyright = True
-                    break
+                    return CheckResult(problems_found=0)
                 if "copyright" in line.lower():
```

Comment:
If you think it's a bug, then go on and create an issue. It was already there, and the purpose of refactoring is not fixing bugs.

Reply:
It seems to me that the intention here is to detect non-standard copyright lines (the check for copyright description is right above).

Reply:
There is also some potential for false positives in this check. I think it may be impossible to reliably detect non-standard ("incorrect" as the log message says) copyright lines.

Reply:
Added issue for multiline copyright, hope some of the issues can be fixed when reworking this feature: https://github.com/Spyro-Soft/scargo/issues/100